r/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM arrow:cpp
+
+# get the stuff
+ADD r /arrow/r
```

Comment:
We don't want to add the whole R directory here, because each change file invalidates the layer cache. 
To further improve the build times We mount the whole arrow directory via the docker-compose volumes configuration. 
> Note this is only suitable because We want "development", not production containers to quickly reproduce possible CI errors.

Reply:
ohhhh ok I was wondering about the way `x-volumes` was being used in the compose file. Makes sense! I can do that